controllers/grafanacontactpoint_controller.go

```diff
@@ -254,7 +234,7 @@ func (r *GrafanaContactPointReconciler) getContactPointFromUID(ctx context.Conte
 func (r *GrafanaContactPointReconciler) finalize(ctx context.Context, contactPoint *grafanav1beta1.GrafanaContactPoint) error {
 	r.Log.Info("Finalizing GrafanaContactPoint")
 
-	instances, err := r.GetMatchingInstances(ctx, contactPoint, r.Client)
+	instances, err := GetAllMatchingInstances(ctx, r.Client, contactPoint)
```

Comment:
Same thing as with https://github.com/grafana/grafana-operator/pull/1815#discussion_r1910215360

Discussing this in the other PR, just commenting to make sure this is handled as well

Reply:
Changed to `GetScopedMatchingInstances` now!